static/elements/chromedash-upcoming.js

```diff
@@ -52,12 +52,110 @@ class ChromedashUpcoming extends LitElement {
       loginUrl: {type: String},
       starredFeatures: {type: Object}, // will contain a set of starred features
       cardWidth: {type: Number},
+      lastFetchedOn: {type: Number},
+      lastMilestoneVisible: {type: Number},
+      milestoneArray: {type: Array},
+      milestoneInfo: {type: Object},
+      cardsToFetchInAdvance: {type: Number},
     };
   }
 
+  set lastFetchedOn(val) {
+    let oldVal = this._lastFetchedOn;
+    this._lastFetchedOn = val;
+    this.fetchNextBatch().then(()=>{
+      this.requestUpdate('lastFetchedOn', oldVal);
+    }).catch(() => {
+      alert('Some error occurred. Please refresh the page or try again later.');
+    });
+  }
+
+  get lastFetchedOn() {
+    return this._lastFetchedOn;
+  }
+
   constructor() {
     super();
     this.cardWidth = this.computeWidthofCard();
+    this.starredFeatures = new Set();
+    this.milestoneArray = [];
+    this.milestoneInfo = {};
+    this.cardsToFetchInAdvance = 3;
+  }
+
+  async fetchNextBatch() {
+    // promise to fetch next milestones
+    const nextMilestones = window.csClient.getSpecifiedChannels(this._lastFetchedOn+1+1,
+      this._lastFetchedOn+1+this.cardsToFetchInAdvance);
+
+    let tempMilestoneArray = [];
```

Comment:
Words like "temp" can almost always be replaced by something more descriptive.  In fact, every variable in a program is temporary to some extent.  Also, this code deals with both milestone numbers and milestone metadata, so it would be good to clarify that this is milestone numbers.   Try something like milestoneNums.

Reply:
Done. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1458/commits/ae2c68acfa46fa965fccb65cc40c52f24ddfff22) commit